aws-crt-kotlin/native/src/aws/sdk/kotlin/crt/io/MutableBufferNative.kt

```diff
@@ -4,33 +4,65 @@
  */
 package aws.sdk.kotlin.crt.io
 
+import aws.sdk.kotlin.crt.Allocator
+import aws.sdk.kotlin.crt.Closeable
+import kotlinx.cinterop.*
+import libcrt.*
+
 /**
  * Represents a mutable linear range of bytes that can be written to.
  * Instance of this class has no additional state except the bytes themselves.
- *
- * NOTE: Platform implementations should provide direct access to the underlying bytes
  */
-public actual class MutableBuffer {
+@OptIn(ExperimentalForeignApi::class)
+public actual class MutableBuffer(buffer: aws_byte_buf? = null, private val capacity: Int) : Closeable { // TODO implement CrtResource?
+    private val buf = buffer ?: Allocator.Default.alloc<aws_byte_buf>()
+
+    public val bytes: ByteArray
+        get() = buf.buffer!!.readBytes(buf.capacity.toInt())
```

Comment:
**discussion**: This function makes a copy of the underlying bytes, I think it's the best we can do with the available CRT APIs but open to suggestions!

Reply:
Why do we need this?

Reply:
From this KDoc in the class:
> NOTE: Platform implementations should provide direct access to the underlying bytes

Do you think we should just not provide access to the underlying bytes?